### PR TITLE
feat(builder): move loop body nodes together when dragging loop [AAP-76892]

### PR DESCRIPTION
**Pull request by skochay330**
## Description

When dragging a Loop node on the workflow builder canvas, loop body and done-branch nodes now move as a group while preserving relative positions. Individual body nodes can still be dragged independently. Loop-back edges also route correctly after save/reopen.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Test coverage improvement

## Changes Made

- [x] Extract shared `loopUtils` (`getLoopBodyMap`, `collectLoopGroupPositions`) used by layout and drag handlers
- [x] Add loop group drag handlers (`onNodeDragStart`, `onNodeDrag`, updated `onNodeDragStop`) in `usePositionEventHandlers`
- [x] Persist group positions from drag offsets (avoids stale `getNodes()` at drag stop)
- [x] Add `useLoopGroupPositionSync` to sync group positions and refresh node internals on workflow load
- [x] Narrow `LoopBackEdge` subscriptions via `useStore` geometry selector
- [x] Add unit tests for loop utils, drag handlers, sync hook, edge geometry, and BuilderFlow wiring
- [x] Add E2E test for loop-back edge rendering after save/reopen

## Out of Scope

- Playwright coverage for drag interactions (headless drag is unreliable; covered by unit/integration tests)
- Migrating legacy workflows with inconsistent stored positions (fixed on next loop drag + save)

## Related Issues

Fixes [AAP-76892](https://issues.redhat.com/browse/AAP-76892)

## How to Test

1. Open the workflow builder and create a Loop (While) with a body script node
2. Drag the Loop node — verify body (and done-branch nodes, if present) move together
3. Drag a single body node — verify only that node moves
4. Save the workflow, reopen it — verify loop-back edge routes correctly without nudging the loop
5. Undo after a loop group drag — verify all positions restore in one step

```bash
cd frontend/packages/syntara-ui && npx vitest run \
  src/routes/builder/utils/loopUtils.test.ts \
  src/routes/builder/hooks/usePositionEventHandlers.test.ts \
  src/routes/builder/hooks/useLoopGroupPositionSync.test.ts \
  src/routes/builder/edges/loopBackEdgeGeometry.test.ts \
  src/routes/builder/edges/LoopBackEdge.test.tsx \
  src/routes/builder/BuilderFlow.test.tsx

cd frontend && npm run e2e -- --grep "loop-back edge renders after save and reopen"
```

## Frontend Checklist

- [x] I have run `npm test` and all tests pass (in `frontend/`)
- [ ] I have run `npm run lint` and code passes all checks
- [ ] I have run `npm run tsc` and there are no type errors
- [x] I have added tests (unit / E2E) for new functionality
- [x] Accessibility has been considered (semantic HTML, labels, keyboard navigation)
- [x] Screenshots or screen recordings are attached for UI changes

## Visual Regression

- [ ] If this PR intentionally changes covered UI: I have commented `/update-screenshots` on this PR and confirmed the regenerated baselines look correct in the Files changed tab
- [ ] If this PR adds a new page/route: I have added an entry to `page-registry.ts` and run `/update-screenshots` to generate its baseline

## General Checklist

- [x] Code follows the project's coding conventions
- [ ] I have updated relevant documentation
- [x] No secrets or credentials are included in this PR
- [ ] Breaking changes are documented with migration steps

## Screenshots / Demo (if applicable)

https://github.com/user-attachments/assets/6497a2f2-d428-4fe6-b406-ee5d0eef58b0

## Additional Notes

Legacy workflows saved before this change may have mismatched loop/body coordinates until the user drags the loop once and re-saves.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        visited.add(nodeId)
+
+        addBodyNode(nodeId)
+
+        // SECURITY: Domain-invariant check — loop body can't exceed total node count
+        if (bodyNodeIds.length > realNodes.length) break
+
+        // Follow source edges, but not edges back to the loop node (prevents circular traversal)
+        const outgoing = realEdges.filter(
+          (e) => e.source === nodeId && e.sourceHandle === EdgeHandleEnum.SOURCE && e.target !== node.id
```
Comment (nicklemmon) [inline frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts:62]: AI-assisted: This code follows edges with the SOURCE handle only. It does not follow the LOOP handle into a nested loop's body. Because of this, `getLoopBodyMap()` may not include the body nodes of an inner loop when it builds the outer loop's body list.

This function drives the new group-drag feature (`usePositionEventHandlers.ts`) and the position sync on load (`useLoopGroupPositionSync.ts`). A drag on an outer loop may not move an inner loop's body nodes with it.

Please test this case: a loop inside a loop, with body nodes in the inner loop. Confirm a drag on the outer loop moves all of these nodes together.

**Review by nicklemmon — COMMENTED** [review]
AI-assisted review.

Note: PR #499 currently bundles this PR's whole loop-grouping feature into its diff, along with an unrelated hook extraction. I left a note there asking the author to confirm that is intended. It does not block this PR.

Good parts:
- Group-drag logic in `usePositionEventHandlers.ts` is well tested: single drag, multi-select drag, done-branch nodes, and a stale-`getNodes()` race at drag stop.
- `loopBackEdgeGeometry.ts` uses a real type guard (`hasPosition`) instead of unsafe casts.
- The `LoopBackEdge` change from `getNodes()` to a reactive `useStore` selector is a real fix, and it is disclosed in the PR description with its own test.
- No new `@ts-ignore`, `@ts-expect-error`, `TODO`, or unsafe `any` casts.

One open question below, left as an inline comment.

File: frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts (hunk 1 above)
Comment (skochay330) [reply frontend/packages/syntara-ui/src/routes/builder/utils/loopUtils.ts:62]: Good catch — confirmed locally. `getLoopBodyMap()` BFS was only following `SOURCE` edges, so when the outer loop reached an inner loop node it stopped there and never entered the inner body via the `LOOP` handle.

Fixed in the latest amend: when BFS visits a nested loop node, it now also queues targets of that loop's `LOOP` handle (and its `DONE` handle when `includeDoneBranch` is true for group drag/sync). Added tests for nested loop body inclusion and nested done-branch handling.

Manually verified: dragging the outer loop now moves the inner loop and its body nodes together.

**Review by nicklemmon — APPROVED** [review]

**Review by mfrances17 — APPROVED** [review]
👍 LGTM, nice work!

**Review by gitdallas — APPROVED** [review]
